sdk/eventhubs/azure_messaging_eventhubs/src/consumer/mod.rs

```diff
@@ -77,23 +81,43 @@ impl ConsumerClient {
     /// use azure_identity::{DefaultAzureCredential, TokenCredentialOptions};
     ///
     ///     let my_credential = DefaultAzureCredential::new()?;
-    /// let consumer = ConsumerClient::new("my_namespace".to_string(), "my_eventhub".to_string(), None, my_credential.clone(), None);
+    /// let consumer = ConsumerClient::builder("my_namespace", "my_eventhub", None, my_credential.clone())
+    ///    .open().await?;
     /// # Ok(())}
     /// ```
-    pub fn new(
+    ///
+    pub fn builder(
+        fully_qualified_namespace: &str,
+        eventhub_name: &str,
+        consumer_group: Option<String>,
+        credential: Arc<dyn TokenCredential>,
```

Comment:
Again, not very builder-y. Or are these all required parameters? The `Option<String>` for `consumer_group` says otherwise.

Reply:
consumer_group is optional, the others are all mandatory.